contributors/design-proposals/nvml-and-volume-support.md

```diff
@@ -0,0 +1,84 @@
+# Proposal for NVML support and nvidia driver volume injection
+We propose leveraging the following to enhance the experimental GPU work:
+*  [NVML library](https://developer.nvidia.com/nvidia-management-library-nvml) to conduct GPU discovery
+* Nvidia driver volume injection
+
+## NVML support and dynamic loading
+Currently, GPU discovery is implemented by querying nvidia GPU devices under
+`/dev/nvidia*` directly, which may bring several problems. For example, it
+cannot detect GPU failures and  it only collects the entry of the device.  We
+think NVML is a more reliable and powerful method for GPU discovery, and advanced  GPU features (driver volume injection, GPU failure detection, heterogeneous GPU, topology-aware scheduling).
+
+We have found that dynamic loading is best when including the NVML library,
+because we can check for its existence at runtime. This allows a single kubelet
+binary to work on nodes that do not include NVML since we can skip all GPU logic.
+
+
+## Nvidia driver volume injection
+ Container-based applications  need Nvidia GPU drivers to interface with the
+ GPU. Installing the  GPU driver in the container is problematic  because the
+ GPU driver version must align with the host GPU driver version exactly for
+ proper functionality. Both host and container libraries need to be updated
+ simultaneously in this deployment.
+
+To simplify, we can borrow an idea from
```

Comment:
Driver volume injection is nearly a MUST for GPU container users and it would be part of kubelet since GPU allocation logic has already been there. We do not want to bring complication to user to start a DaemonSet before they use the very basic GPU feature.